### DISCUSSION TITLE
[cli] Option to disable launching storybook after installation

### DISCUSSION THREAD

**Discussion by yusitnikov** [storybookjs/storybook#25510] [Ideas]
### Is your feature request related to a problem? Please describe.

Hello! I'm working on a script that automatically creates a React project with Storybook, and the way to add Storybook to the project is to run `npx storybook@latest init --yes`. Unfortunately, this command always launches storybook after installing, which is not appropriate for a CLI script that is supposed to do other things after installing Storybook.

### Describe the solution you'd like

Suggest adding an option to the CLI script that will disable launching Storybook automatically, e.g. `npx storybook@latest init --no-launch`

### Describe alternatives you've considered

Another option I considered is to import `@storybook/cli` package as an npm dependency in the script, and use its API to install Storybook without launching it afterwards. Unfortunately, the package doesn't provide a public API for that, and generally doesn't look like it's supposed to be used this way.

### Are you able to assist to bring the feature to reality?

no

**Comment by Turkitutu** [comment]
I really need this feature

**Comment by joeslazaro** [comment]
What I did for my own Storybook automation project is to skip init and instead just install the packages that it normally installs, and then set up the script to run `storybook build`. It works fine for me since I don't need support for any of the init CLI flags, like changing the parser, builder, or package manager.

**Comment by valentinpalkovic** [comment]
You can see [here](https://github.com/storybookjs/storybook/blob/next/code/lib/cli/src/initiate.ts#L408) that the dev command only executes when the env variables `CI` and `IN_STORYBOOK_SANDBOX` aren't set to `'true'`. Setting one of the env variables to 'true' should skip starting Storybook after init.

**Reply by fastfrwrd** [reply]
This workaround definitely does work, but it does seem that it would be better to have an explicit option that does this, since the `CI` environment variable has a well-defined meaning that may or may not apply to the situation we want to use this feature in.

For example, I want to invoke the storybook CLI on behalf of a user as part of a project scaffold tool that needs to run the storybook CLI in a spawned process and then apply a codemod to the files storybook created; that is all happening on a user's machine, presumably, so it's a smell to me that the only way we could opt into that is to set the `CI` environment variable, which is meant to indicate that we're _not_ on a user's machine.

My ideal version of this (which I think I'll create a PR for) would have the same default behavior as now, with the added optional CLI option of `--no-dev`, which takes precedence if set (similar to the `--no-open` option on `start`).

**Reply by valentinpalkovic** [reply]
That makes sense to me. Would you like to provide a PR with the implementation?

**Reply by fastfrwrd** [reply]
opened https://github.com/storybookjs/storybook/pull/26918

**Comment by yusitnikov** [comment]
The `CI` env var worked like a charm, thank you!

**Reply by valentinpalkovic** [reply]
@yusitnikov The `--no-dev` flag was implemented and you can use it as documented here: https://storybook.js.org/docs/api/cli-options
